components/script/dom/document.rs

```diff
@@ -1998,7 +2002,7 @@ impl Document {
     }
 
     pub fn new(window: &Window,
-               browsing_context: Option<&BrowsingContext>,
+               has_browsing_context: bool,
```

Comment:
I lean towards a self-documenting enum here, given that every caller passes a boolean literal.